ribasim_qgis/scripts/qgis_testrunner.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+
+"""
+***************************************************************************
+    Launches a unit test inside QGIS and exit the application.
+
+    Arguments:
+
+    accepts a single argument with the package name in python dotted notation,
+    the program tries first to load the module and launch the `run_all`
+    function of the module, if that fails it considers the last part of
+    the dotted path to be the function name and the previous part to be the
+    module.
+
+    Extra options for QGIS command line can be passed in the env var
+    QGIS_EXTRA_OPTIONS
+
+    Example run:
+
+    # Will load geoserverexplorer.test.catalogtests and run `run_all`
+    QGIS_EXTRA_OPTIONS='--optionspath .' \
+        GSHOSTNAME=localhost \
+        python qgis_testrunner.py geoserverexplorer.test.catalogtests
+
+
+    GSHOSTNAME=localhost \
+         python qgis_testrunner.py geoserverexplorer.test.catalogtests.run_my
+
+
+    ---------------------
+    Date                 : May 2016
+    Copyright            : (C) 2016 by Alessandro Pasotti
+    Email                : apasotti at boundlessgeo dot com
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
```

Comment:
I need to add information here that I changed the file so that it works on Windows and Linux. Add the commit hash.

Also, what about this specific license. Am I allowed to commit it like this?

Reply:
We don't distribute this file, we only use it to run tests. It is therefore fine to use it.

If I remember correctly, your current code barely resembles that original testrunner and only uses a few general concepts like monkey patching QGIS. IMO, you can just start a new file without mentioning any copyright information then.